KarhooUISDK/Screens/CheckoutScreen/Components/CheckoutViews/KarhooCheckoutHeaderView.swift

```diff
@@ -118,9 +118,8 @@ final class KarhooCheckoutHeaderView: UIStackView {
         return view
     }()
     
-    private lazy var learnMoreButton: KarhooLearnMoreButton = {
-        let button = KarhooLearnMoreButton()
-        button.set(actions: self)
+    private lazy var learnMoreButton: KarhooExpandViewButton = {
+        let button = KarhooExpandViewButton(title: UITexts.Booking.learnMore, onExpand: learnLessPressed, onCollapce: learnMorePressed)
```

Comment:
I like this syntax very much, but unfortunately this way we are creating a memory leak since the closure is escaping. Unless something changed (I hope so!) in swift the closure mechanism, you need to use `weak` somewhere.